dsl_parser/functions.py

```diff
@@ -308,9 +308,13 @@ def get_node_template(self, handler):
                     '{0} can only be used within a relationship but is used '
                     'in {1}'.format(self.node_name, self.path))
             if self.node_name == SOURCE:
-                node = self.context['node_template']
+                # this is messy, see CY-1676 to fix it
+                node = self.context.get('node_template') or \
```

Comment:
maybe put some doc also here, so it will be more clear without the Jira

Reply:
what can be more clear than this? it's messy, gotta clean it up, the jira is about cleaning it up